### ISSUE TITLE
pretrained_models

### ISSUE THREAD

**Issue by xinli2008** [aa1234241/vqgan#1]
Hi, nice job, can you share the pretrained generator and discriminator models ?
`torch.load(os.path.join("/media/userdisk1/code/VQGAN-pytorch/checkpoints", "discriminator_epoch_308.pt")))`

Best

**Comment by aa1234241** [comment]
Sure, You can access it [here](https://pan.baidu.com/s/1vEHsRV1XFbIvu1VJ9b2Y9g?pwd=fpd3). By the way, could you help me out by uploading some of the generated results under this thread?

**Comment by xinli2008** [comment]
Thank you for you nice reply and i have access to the model. Besides, when i complete the training and inference process and i will upload the generated imagess
Best

**Comment by xinli2008** [comment]
Hello, I’m curious if the weights you provided are fully compatible with the model architecture in your repository. I followed your code and typed it out almost line by line, but I keep encountering a mismatch when I try to load your pretrained weights during training.
best

**Comment by aa1234241** [comment]
Hello, could you show me the error message?

**Comment by xinli2008** [comment]
I think the issue might be caused by me manually naming something differently from yours. I noticed that the model weights I saved are the same as yours, so I'll check it myself first.

**Comment by xinli2008** [comment]
I rechecked the code, and the reason the weights weren't properly perturbed is that some of the parameter definitions in my code differ from yours, which caused a loading error. I've made the necessary adjustments, and now the weights load correctly.

Secondly, I noticed that during training, the `discriminator_loss` remains zero due to the effect of `self.vqgan.adopt_weight`. Will this cause any issues?

**Comment by aa1234241** [comment]
> I rechecked the code, and the reason the weights weren't properly perturbed is that some of the parameter definitions in my code differ from yours, which caused a loading error. I've made the necessary adjustments, and now the weights load correctly.
> 
> Secondly, I noticed that during training, the `discriminator_loss` remains zero due to the effect of `self.vqgan.adopt_weight`. Will this cause any issues?

No, the discriminator_loss is the GAN loss, and it is deliberately set to be zero for some iterations, this helps the model training.

**Comment by xinli2008** [comment]
Hello, sorry to bother you. I wrote a test and visualization code for your pre-trained weights according to the forward process in your data loading and training, but the results always seem to be incorrect. If you have time, can you help me check it?
```python
import os
import torch
import numpy as np
from PIL import Image
import argparse
import albumentations
from vqgan import VQGAN

class TestModel:
    def __init__(self, model, device):
        self.model = model.to(device)
        self.device = device

    def load_image(self, image_path):
        image = Image.open(image_path).convert("RGB")
        return np.array(image).astype(np.float32) / 255.0  

    def preprocess_image(self, image):
        rescaler = albumentations.SmallestMaxSize(max_size=256)
        cropper = albumentations.CenterCrop(height=256, width=256)
        preprocessor = albumentations.Compose([rescaler, cropper])

        image = preprocessor(image=image)["image"]
        image = (image / 127.5 - 1.0).astype(np.float32)  
        image = image.transpose(2, 0, 1)
        return torch.tensor(image).unsqueeze(0).to(self.device)

    def test_and_save_results(self, input_folder, output_folder):
        if not os.path.exists(output_folder):
            os.makedirs(output_folder)

        for image_name in os.listdir(input_folder):
            image_path = os.path.join(input_folder, image_name)
            if os.path.isfile(image_path):
                # Load and preprocess image
                input_image = self.load_image(image_path)
                input_tensor = self.preprocess_image(input_image)

                with torch.no_grad():
                    decoded_images, _, _ = self.model(input_tensor)

                output_image = ((decoded_images.squeeze(0).cpu().numpy().transpose(1, 2, 0) + 1) * 127.5).clip(0, 255).astype(np.uint8)
                output_image_pil = Image.fromarray(output_image)
                output_image_pil.save(os.path.join(output_folder, image_name))

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description = "VQGAN Training Parameters")
    parser.add_argument("--latent_dim", type = int, default = 256, help = "latent dimension of vqgan")
    parser.add_argument("--num_codebook_vectors", type = int, default = 1024, 
                        help = "number of vqgan codebook vectors")
    parser.add_argument("--beta", type = float, default = 0.25, help = "commitment loss scalar")
    parser.add_argument("--image_channels", type = int, default = 3, help = "number of channels of images")
    parser.add_argument("--device", type = str, default = "cuda", help = "which device to train the data, cuda or cpu")
    args = parser.parse_args()
    model = VQGAN(args).to(device = args.device)
    model.load_state_dict(torch.load(os.path.join("/mnt/VQGAN_from_scratch/pretrained_models", "vqgan_epoch_308_flower.pt")))
    tester = TestModel(model, args.device)
    tester.test_and_save_results("/mnt/VQGAN_from_scratch/input", "/mnt/VQGAN_from_scratch/output")

**Comment by aa1234241** [comment]
Hello, you can try to directly run the ```training_vqgan.py``` with the pre-trained weights in this repo, and it will generate results in the ```.\results``` folder. I just downloaded this repo and directly ran it.  The results look like this:
![309_0](https://github.com/user-attachments/assets/ef93c782-cd88-41e1-bc46-c7ffe5de4e8e) Top row shows the original images, and the bottom row shows the reconstructed images. Make sure this step generates the correct results, then you can find the bugs of your code.

**Comment by xinli2008** [comment]
![image](https://github.com/user-attachments/assets/99250705-2594-48ec-93a6-a05187f8b7d4)
Thank you for your advice and the results seems wrong. And it means that the mistake happens in my code of the model rather than the visualize code. I will try to find the mistake by comparing the code.
Best

**Comment by aa1234241** [comment]
> ![image](https://private-user-images.githubusercontent.com/71311087/376279514-99250705-2594-48ec-93a6-a05187f8b7d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MTY3MjcsIm5iZiI6MTcyODkxNjQyNywicGF0aCI6Ii83MTMxMTA4Ny8zNzYyNzk1MTQtOTkyNTA3MDUtMjU5NC00OGVjLTkzYTYtYTA1MTg3ZjhiN2Q0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDE0MzM0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0OWRkMjYyN2FlNjc4YmE5YjdlZWNhZDg3NTFiODk5MmVlOGJkZjljMjdkY2QwNWJkNDk1ODA5YTQ4ZGI0MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dLHgZYRWxliqx7Hrer3bP2KVwzNxM8BTVDo-BDo69tI) Thank you for your advice and the results seems wrong. And it means that the mistake happens in my code of the model rather than the visualize code. I will try to find the mistake by comparing the code. Best

seems like the pretained weights are not correctly loaded. You can first try to use the code in this repo, and then debug your code

**Comment by xinli2008** [comment]
thank you for your advice and i will try it. Keep in touch.
